lib/cascading-config-array-factory.js

```diff
@@ -218,17 +223,17 @@ class CascadingConfigArrayFactory {
         builtInRules = new Map(),
         loadRules,
         resolver,
-        eslintRecommendedPath,
-        eslintAllPath
+        eslintRecommendedConfig,
+        eslintAllConfig
```

Comment:
As an alternative, we could keep these string options, add new ones for objects, and support both paths and objects (if the objects are passed in, then use them; otherwise, use paths). That wouldn't be a breaking change for this package. Though it would make the code a bit more complicated, and I'm not sure if anyone other than eslint/eslint is using this API.

@nzakas thoughts about this, should we keep the existing API and just add new options?

Reply:
Yeah, I don’t want to make any breaking changes at this point. It probably makes sense to just add new properties and throw an error if both a path and an object are passed.

Reply:
Thanks for the comments!

I've updated the code to keep the compatibility.